### PR TITLE
Reduce redundant LYS option API Calls 

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We're fetching the LYS options on every WC screen, which is unnecessary. We only need to fetch the LYS options on the home screen, and this PR restricts the API calls to that screen only.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site
2. Open dev tools and go to the network tab and filter by `woocommerce_coming_soon`
3. Visit the home page and check if the API call is made
4. Visit other pages and check if the API call is not made

### Changelog entry

- [ ] Automatically create a changelog entry from the details below.

- [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

- [ ] Patch
- [ ] Minor
- [ ] Major

#### Type

- [ ] Fix - Fixes an existing bug
- [ ] Add - Adds functionality
- [ ] Update - Update existing functionality
- [ ] Dev - Development related task
- [ ] Tweak - A minor adjustment to the codebase
- [ ] Performance - Address performance issues
- [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/activity-panel/activity-panel.js [hunk 1]
```diff
@@ -65,6 +65,10 @@ const SetupTasksPanel = lazy( () =>
 );
 
 export const ActivityPanel = ( { isEmbedded, query } ) => {
+	const isHomescreen = () => {
+		return query.page === 'wc-admin' && ! query.path;
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/activity-panel/activity-panel.js]: Does this need to be a function?

**Review by rjchow — APPROVED** [review]
Works as described! I do have one suggestion for safety - instead of returning only nulls I think it would be good to specifically indicate that fetching was disabled, and also accept and indicate this state visually in the badge. This hints to the future developer that there this state exists

**Comment by chihsuan** [comment]
> Works as described! I do have one suggestion for safety - instead of returning only nulls I think it would be good to specifically indicate that fetching was disabled, and also accept and indicate this state visually in the badge. This hints to the future developer that there this state exists

Thanks for the suggestion! That does sound safer. I'll add a follow-up issue for it. (Added: https://github.com/woocommerce/woocommerce/issues/48307)

File: plugins/woocommerce-admin/client/activity-panel/activity-panel.js (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/activity-panel/activity-panel.js]: Nope, we can change it to a constant. 👍

Comment (chihsuan) [reply plugins/woocommerce-admin/client/activity-panel/activity-panel.js]: Changed in 3de26547516666b1a325132740db4b3c67858c54

**Review by psealock — APPROVED** [review]
Code looks great, I like that it defaults to `true`. Nice one 🚢

**Pull request merged**